app/process_proposal.go

```diff
@@ -68,6 +68,11 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) (resp abci.Resp
 		}
 
 		sdkTx, err := app.txConfig.TxDecoder()(tx)
+		// Set the tx bytes in the context for app version v3 and greater
+		if sdkCtx.BlockHeader().Version.App >= 3 {
+			sdkCtx = sdkCtx.WithTxBytes(tx)
+		}
```

Comment:
same nit here: this would be easier to follow if it was placed right after sdkCtx is initialised on line 54

Reply:
it needs to be done while we're iterating over transactions in later lines no?

Reply:
so you suggest i iterate over block txs another time after context is created and set `WithTxBytes` there?

Reply:
ignore me. I misunderstood this